File: dialogio/sdl/pas/valid_drive.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

function valid_drive(drive: Char; var info: String): Boolean;

var
  idx: Byte;

begin
  valid_drive := FALSE;
  info := '';
  {$IFNDEF UNIX}
  idx := 0;
  For idx := 0 to 128 do
    If (drive_list[idx] = drive) then
      begin
        info := 'DRiVE';
        BREAK;
      end;
  {$ENDIF}
  If (info <> '') then valid_drive := TRUE;
end;

File: dialogio/sdl/valid_drive.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

static bool valid_drive (__UNUSED char drive, String *info)
{
  SetLength (info, 0);

#if GO32 || WIN32 || WIN64 || WINNT
  for (uint8_t i = 0; i < 129; i++)
    if (drive_list[i] == drive)
    {
      CopyString (info, (String *)"\x05" "DRiVE", 255);
      break;
    }
#endif // GO32 || WIN32 || WIN64 || WINNT

  return Length (info) != 0;
}
